### PR TITLE
docs: fix typos and add some links for README.md

### LINKED ISSUE #17: docs: typos in README.md
There are a few typos in README.md

Details can be found in: #16

### PR DESCRIPTION
**Pull request by hk-mp5a3**
# Description
Fix typos and add some links for README.md

# Type of Change:
Documentation

# How Has This Been Tested?
There are no changes in code files, and the only changes are in documentations.

### CODE REVIEW

**Review by yifeili98 — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -49,10 +49,10 @@ We use [pylint](https://pylint.org) as Python linter.
 
 ## Coding Style
 
-In general, we follow PEP8 Python coding style and Google pyguide
-However, if PEP8 and Google pyguide has conflict style, PEP8 take place.
+In general, we follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) Python coding style and Google pyguide
```
Comment (yifeili98) [inline README.md]: Should we add a `.` at the end?
`In general, we follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) Python coding style and Google pyguide.`

File: README.md [hunk 2]
```diff
@@ -37,10 +37,10 @@ npm install
 pip install -r requirements.txt
 ```
   
-We require all contributor write docsting for other contributors easy to follow.  
-One of software development process is TDD. We highly recommended you *write test before code*  
-However, test is not require but recommended.  
-The philosophy is **you must know exactly the detailed behavior of your code doing**.  
+We require all contributors to write docsting for other contributors easy to follow.  
+One of the software development process is TDD(Test-driven Development). We highly recommended you *write at least one test before code*
```
Comment (yifeili98) [inline README.md]: Should we change `process` to `processes`?
`One of the software development processes is TDD(Test-driven Development). We highly recommended you *write at least one test before code*.`

File: README.md [hunk 3]
```diff
@@ -49,10 +49,10 @@ We use [pylint](https://pylint.org) as Python linter.
 
 ## Coding Style
 
-In general, we follow PEP8 Python coding style and Google pyguide
-However, if PEP8 and Google pyguide has conflict style, PEP8 take place.
+In general, we follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) Python coding style and Google pyguide
+However, if [PEP 8](https://www.python.org/dev/peps/pep-0008/) and [Google Python Style Guide](http://google.github.io/styleguide/pyguide.html) have conflict style, [PEP 8](https://www.python.org/dev/peps/pep-0008/) take place.
```
Comment (yifeili98) [inline README.md]: Should we change `take` to `takes`?
`[PEP 8](https://www.python.org/dev/peps/pep-0008/) takes place.`

Comment (yifeili98) [reply README.md]: Also, can be rephrased to 
`However, when [PEP 8](https://www.python.org/dev/peps/pep-0008/) is in conflict with [Google Python Style Guide](http://google.github.io/styleguide/pyguide.html), [PEP 8](https://www.python.org/dev/peps/pep-0008/) should be given precedence.`

File: README.md [hunk 4]
```diff
@@ -37,10 +37,10 @@ npm install
 pip install -r requirements.txt
 ```
   
-We require all contributor write docsting for other contributors easy to follow.  
-One of software development process is TDD. We highly recommended you *write test before code*  
-However, test is not require but recommended.  
-The philosophy is **you must know exactly the detailed behavior of your code doing**.  
+We require all contributors to write docsting for other contributors easy to follow.
```
Comment (yifeili98) [inline README.md]: `docsting` to `docstring`

Comment (hk-mp5a3) [reply README.md]: Good point! Resolved.

File: README.md (hunk 2 above)
Comment (hk-mp5a3) [reply README.md]: Good point! Resolved.

File: README.md (hunk 1 above)
Comment (hk-mp5a3) [reply README.md]: Good point! Resolved.

File: README.md (hunk 3 above)
Comment (hk-mp5a3) [reply README.md]: Good point! Resolved.

File: README.md [hunk 5]
```diff
@@ -37,8 +37,8 @@ npm install
 pip install -r requirements.txt
 ```
   
-We require all contributors to write docsting for other contributors easy to follow.  
-One of the software development process is TDD(Test-driven Development). We highly recommended you *write at least one test before code*  
+We require all contributors to write docstring for other contributors easy to follow.
```
Comment (yifeili98) [inline README.md]: May want to rephrase to
`We require all contributors to write docstring for other contributors so the codes are easy to follow.  `

File: README.md [hunk 6]
```diff
@@ -37,8 +37,8 @@ npm install
 pip install -r requirements.txt
 ```
   
-We require all contributors to write docsting for other contributors easy to follow.  
-One of the software development process is TDD(Test-driven Development). We highly recommended you *write at least one test before code*  
+We require all contributors to write docstring for other contributors easy to follow.  
+One of the software development processes is TDD(Test-driven Development). We highly recommended you *write at least one test before code*
```
Comment (yifeili98) [inline README.md]: nit: One of the software development processes is `called` TDD?

File: README.md (hunk 5 above)
Comment (hk-mp5a3) [reply README.md]: resolved.

**Review by yifeili98 — APPROVED** [review]
Awesome, LGTM!

**Review by kis87988 — APPROVED** [review]
LGTM

**Pull request merged**